### ISSUE TITLE
Background color for neutral code snippets is too similar to bad one

### ISSUE THREAD

**Issue by kwalrath** [dart-lang/site-www#1093]
More info: https://github.com/dart-lang/site-www/pull/1050

And the original report:

> For example, code snippets under https://www.dartlang.org/guides/language/effective-dart/usage#dont-use-listfrom-unless-you-intend-to-change-the-type-of-the-result looks to have similar background color as the code snippets annotated as "bad".
> 
> I was wondering why this code was bad, and only after careful inspection, realized that the "bad" from the top-right corner was missing. Consider using a background color that is different from the "bad" and "good" snippets .
>

**Comment by orcutt989** [comment]
I'm not seeing a class for the neutral code block. I see the CSS class for "good" and "bad" and their corresponding colors, but not for the neutral.  I assume neutral means, what I would call, the grey `#eee` boxes? That looks like its defined by the `.prettyprint` cass or `pre {` tag. Perhaps I'm wrong and the "neutral" boxes are the `#fcf8e3` "yellow" boxes defined by the `.alert.alert-warning` class?

**Comment by chalin** [comment]
I would tent to disagree that colors need to be adjusted, here's why. Consistent with good site design principles, we don't use color as the sole mechanism to convey semantic information. That is, we could fall back to grayscale, and the pages would still be meaningful. Specifically: in Effective Dart, "good"/"bad" code excerpts are explicitly labeled with the text "good"/"bad". Regular code excerpts have no such label.

Bob "[turned] a couple of the neutral examples into explicit good and bad ones", and the regular code excerpts that remain seem to be clearly "regular" from context.

That being said, I'm not against tweaking the colors, but it can be tricky, and requires testing across different displays of varying resolution (for example, I've noticed that certain opacity settings are lost of low res displays).

(Finally, we _might_ be adjusting code excerpt formatting across the board in the not too distant future.)

**Comment by orcutt989** [comment]
@chalin I think that's a good point, but I am just trying to figure out which color we are talking about is "neutral".

**Comment by kwalrath** [comment]
"neutral" is the gray background. But now that @chalin brings it up, I agree that perhaps we should table this because it looks like we'll have a designer look at our sites soon. I'm going to mark this as blocked, for now.

**Comment by tijoforyou** [comment]
Hi, I was the one who originally suggested to adjust the color a little. As a color-blind person, it is a little hard to quickly figure out which-is-which.

That said, I agree that color should not be the only indication to convey this information.
But, the current "good"/"bad" annotations doesn't really help here either. Those have poor contract and is much easier to miss. For example, see the attached picture. Most of the main contents are flushed towards the left side, which means, I can easily miss those annotations, unless I know to look for those specifically.

![effective-dart-colors](https://user-images.githubusercontent.com/14226604/67442672-bde0f380-f5b6-11e9-941c-95b739a5c626.png)

If the background colors cannot be adjusted, can we adjust the position/styling/contrast of "good"/"bad" so that is more easily visible?

**Comment by kwalrath** [comment]
Thanks for the followup, @tijoforyou.

**Comment by chalin** [comment]
Yes, thanks for the detailed followup. Maybe we can left-justify the good/bad labels. WDYT?

**Comment by kwalrath** [comment]
> Yes, thanks for the detailed followup. Maybe we can left-justify the good/bad labels. WDYT?

The new design does have a lot of space around the code boxes now... It seems like we could put the labels at the upper left.

**Comment by tijoforyou** [comment]
Moving to upper left would be great IMO. That way I see this annotation first, and then can update my mental model to know what kind of example follows.

**Comment by chalin** [comment]
Hmm, I'm not sure. @kwalrath WDYT?

> ![image](https://user-images.githubusercontent.com/4140793/67532382-03272300-f694-11e9-9715-471d620c53a2.png)

Maybe adding a bounding box around the label might help?

> ![image](https://user-images.githubusercontent.com/4140793/67533011-7df13d80-f696-11e9-9b93-1cb25ab63cda.png)

Or with a thinker border:

> ![image](https://user-images.githubusercontent.com/4140793/67532969-539f8000-f696-11e9-9c6a-94fd822490a0.png)

**Comment by kwalrath** [comment]
I do like having some kind of border/separation, but don't want it to be confused for a button.

I can live with #2 (smaller border), and if you stage it, then we can play with it and see if we can come up with something even better.

**Comment by laurameck** [comment]
Hi, is this available to pick up?

**Comment by chalin** [comment]
As far as I know this is up for grabs, so ... it's yours! 😄  (If you'd like, I can try to dig up the style changes I used to create the screenshots shown above, but if so, that won't happen until early next week.)

**Comment by laurameck** [comment]
Could someone with color blindness also weigh in on if this is enough of an improvement? I shifted the good/bad to the top left like @chalin, but I also increased the font size to match the code snippet to hopefully make it more legible. I can certainly add a border if the size and location change isn't sufficient.
<img width="954" alt="Screen Shot 2020-10-02 at 9 33 50 PM" src="https://user-images.githubusercontent.com/9971478/94982379-32c2ea00-04f7-11eb-976f-b15b68c35bfa.png">
